src/app/clusters/valve-configuration-and-control-server/valve-configuration-and-control-server.cpp

```diff
@@ -376,14 +376,28 @@ CHIP_ERROR UpdateCurrentLevel(EndpointId ep, Percent currentLevel)
     if (HasFeature(ep, ValveConfigurationAndControl::Feature::kLevel))
     {
         VerifyOrReturnError(Status::Success == CurrentLevel::Set(ep, currentLevel), CHIP_IM_GLOBAL_STATUS(ConstraintError));
-        return CHIP_NO_ERROR;
     }
-    return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+    DataModel::Nullable<Percent> targetLevel = DataModel::NullNullable;
+    TargetLevel::Get(ep, targetLevel);
+    if (!targetLevel.IsNull() && currentLevel == targetLevel.Value())
+    {
+        targetLevel = DataModel::NullNullable;
```

Comment:
Spec currently doesn't say TargetLevel should be nulled once opening or closing is complete. It might be important not to explicitly null it, since it allows to verify the target and current level are similar for troubleshooting purposes. It could also be used, in addition to the CurrentState, to verify state of change.

Reply:
This change also causes VALCC_3_3 to fail.

Reply:
The description of the TargetState attribute does IMO indicate that it is expected to be nulled when the transition is done, this might now be when Open and Close are "done" but instead when the transition is done.